woocommerce/woocommerce-android
# [Hackweek] Customer details screen

⚠️ This PR depends on [this FluxC PR](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/3040)

### Why
We want to match the iOS experience and include a customer section in the Hub Menu
pe5pgL-4L1-p2

### Description
This PR focuses on displaying the customer details for the Customers section in the menu hub. The screen is receiving data from two different sources: the customer's endpoint and the reports (analytics) endpoint. Every time a merchant navigates to the screen, we refresh the local customer's data and display the customer's data from the DB. The `GetCustomerWithStats` will be in charge of getting the customer data from two different sources and merging them to retrieve the best possible outcome. The `RefreshCustomerData` use case will refresh the customer's DB data.

### Testing information
TC1
1. Open the app
2. Tap o Menu
3. Check that now there is a Customers section
4. Tap on the Customers section
5. Check that the app displays the store's customers information
6. Tap on a customer
7. Check that the app navigates to customer details and displays the customer information

### Images/gif

https://github.com/woocommerce/woocommerce-android/assets/18119390/bed1c291-045c-433a-bd16-ec4d41cfc7b2
